### PR TITLE
Show the correct number of providers next to each user when a provider is managing users

### PR DESCRIPTION
**Pull request by steventux**
## Context

We restrict the operations the current provider user can perform based on their provider permissions.
When they view the users index, the provider information for each user should reflect the same constraints on what provider information is visible.

## Changes proposed in this pull request

Scope providers to those manageable by current provider user when displaying provider information on the provider users index.

## Link to Trello card

https://trello.com/c/dQnskexv/2069-show-the-correct-number-of-providers-next-to-each-user-when-a-provider-is-managing-users

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/models/provider.rb [hunk 1]
```diff
@@ -26,6 +26,11 @@ class Provider < ApplicationRecord
 
   audited
 
+  scope :manageable_by, ->(provider_user) do
```
Comment (duncanjbrown) [inline app/models/provider.rb:29]: Is this technically `with_users_manageable_by`?

File: app/components/provider_interface/user_list_card_component.rb [hunk 2]
```diff
@@ -2,16 +2,18 @@ module ProviderInterface
...
+    def initialize(provider_user:, manageable_providers:)
       @provider_user = provider_user
+      @manageable_providers = manageable_providers
       @full_name = provider_user.full_name
       @email_address = provider_user.email_address
     end
 
     def providers_text
-      providers = provider_user.providers
+      providers = provider_user.providers.where(id: manageable_providers.pluck(:id)).order(:name)
```
Comment (duncanjbrown) [inline app/components/provider_interface/user_list_card_component.rb:15]: I think the `pluck` will run on each row render — could we push it up into the controller so it only runs once? Maybe the controller could generate structs or hashes like `{ user: user, providers: user.providers.where.etc }` and the component could be dumber

File: app/models/provider.rb (hunk 1 above)
Comment (steventux) [reply app/models/provider.rb]: Amended here: https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2022/commits/c76fe3e881ac35c3538a122dfc753425f8ae8eb3

File: app/components/provider_interface/user_list_card_component.rb (hunk 2 above)
Comment (steventux) [reply app/components/provider_interface/user_list_card_component.rb]: I've done something like that here https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2022/commits/d2e2a0ffaedccab44e33b08b12de803d3b8bd09f

**Pull request merged**